### PR TITLE
[js/web] set noUnusedParameters to true and fix a few bugs

### PR DESCRIPTION
**Pull request by fs-eire**
### Description
- set tsconfig "noUnusedParameters" to `true` and fix a few bugs discovered by typescript.
   how unused parameter is fixed:
   - for most code (webgl), add underscore as prefix, which is the standard ignore pattern for typescript check.
   - remove unused parameter from function and modify corresponding function calls (jsep)
   - fix a bug in ArgMinMax: this 2 operators do not have more than one input(s) so the `createArgMinMaxAttributesFromInputs()` is removed.
- add proxy main.ts into typescript check and fix a bug in parameter passing
   - fixed `run()` function call and add typecheck fix (hack)

### CODE REVIEW

File: js/web/lib/wasm/proxy-worker/main.ts [hunk 1]
```diff
@@ -3,15 +3,47 @@
 
 /// <reference lib="webworker" />
 
-import {OrtWasmMessage} from '../proxy-messages';
+//
+// * type hack for "HTMLImageElement"
+//
+// in typescript, the type of "HTMLImageElement" is defined in lib.dom.d.ts, which is conflict with lib.webworker.d.ts.
+// when we use webworker, the lib.webworker.d.ts will be used, which does not have HTMLImageElement defined.
```
Comment (satyajandhyala) [inline js/web/lib/wasm/proxy-worker/main.ts:10]: This description is confusing. If HTMLImageElement is not defined in lib.webworker.d.ts then why there is a conflict. I understand that there is a good reason to make this change but the description is not clear.

**Comment by satyajandhyala** [comment]
Is the leading '_' only required for the first parameter?

File: js/web/lib/wasm/proxy-worker/main.ts (hunk 1 above)
Comment (satyajandhyala) [reply js/web/lib/wasm/proxy-worker/main.ts:10]: Is it possible to add a library as suggested [here](https://stackoverflow.com/questions/42603783/missing-basic-dom-types-in-typescript-project)?

**Comment by fs-eire** [comment]
> Is the leading '_' only required for the first parameter?

the leading '_' tells typescript that I know this parameter is not used and I intend to do so. So typescript will no longer warn on that parameter. it does not need to be the first parameter.

**Review by satyajandhyala — APPROVED** [review]

**Pull request merged**